src/NFTAuction.sol

```diff
@@ -75,15 +56,26 @@ contract NFTAuctionMachine is
             address(this)
         )
     {
-        if (_duration == 0) {
-            revert INVALID_DURATION();
-        }
+        nft = _nft;
         auctionDuration = _duration;
-        auctionEndingAt = block.timestamp + _duration;
         projectId = _projectId;
-        metadata = _metadata;
         weth = _weth;
-        maxSupply = _maxSupply;
+    }
+
+    /**
+    @dev Transfers eth/weth.
+    */
+    function _transferFunds(address _bidder, uint256 _amount) internal {
+        if (_amount > 0) {
+            (bool sent, ) = _bidder.call{value: _amount, gas: 20000}("");
+            if (!sent) {
+                weth.deposit{value: _amount}();
+                bool success = weth.transfer(_bidder, _amount);
+                if (!success) {
+                    revert TOKEN_TRANSFER_FAILURE();
```

Comment:
should revert? will kill auction forever if it fails. maybe just return false instead?

Reply:
it can only fail if there is no balance in the contract because now for eth transfer we also have a limit let me think more about this

Reply:
@nnnnicholas the thing is if we return false the prev. highest bidder won't be able to get the funds back as the storage is overwritten so I think better to keep with the gas limit and a fallback on weth transfer we should be good IMO